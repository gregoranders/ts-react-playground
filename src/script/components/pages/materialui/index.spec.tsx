import React, { mount, create, RecoilRoot } from '@app/testUtils';

import * as TestSubject from '@pages/materialui';

describe(`${TestSubject.MaterialUIPage.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    expect(testSubject.text()).toBe('HideMaterialUIPage');
  });

  describe('interaction', () => {
    it('should contain `Hide` in initial state', () => {
      const testSubject = mount(
        <RecoilRoot>
          <TestComponent />
        </RecoilRoot>,
      );
      expect(testSubject.text()).toBe('HideMaterialUIPage');
    });
    it('should contain `Show` when clicked', () => {
      const testSubject = mount(
        <RecoilRoot>
          <TestComponent />
        </RecoilRoot>,
      );
      testSubject.find('button').simulate('click');
      expect(testSubject.text()).toBe('Show');
    });
  });
});
