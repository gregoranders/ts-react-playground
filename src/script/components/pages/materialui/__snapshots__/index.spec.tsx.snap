// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialUIPage snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <span
    className="MuiSwitch-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiSwitch-switchBase MuiSwitch-colorSecondary"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6 MuiSwitch-input"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      className="MuiSwitch-track"
    />
  </span>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Hide
    </span>
  </button>
  <h2
    className="MuiTypography-root MuiTypography-h1"
  >
    MaterialUIPage
  </h2>
</div>
`;
